src/ray/common/asio/periodical_runner.cc

```diff
@@ -42,40 +43,43 @@ void PeriodicalRunner::RunFnPeriodically(std::function<void()> fn, uint64_t peri
         [this, fn = std::move(fn), period_ms, name, timer = std::move(timer)]() {
           if (RayConfig::instance().event_stats()) {
             DoRunFnPeriodicallyInstrumented(
-                fn, boost::posix_time::milliseconds(period_ms), *timer, name);
+                fn, boost::posix_time::milliseconds(period_ms), timer, name);
           } else {
-            DoRunFnPeriodically(fn, boost::posix_time::milliseconds(period_ms), *timer);
+            DoRunFnPeriodically(fn, boost::posix_time::milliseconds(period_ms), timer);
           }
         },
         "PeriodicalRunner.RunFnPeriodically");
   }
 }
 
-void PeriodicalRunner::DoRunFnPeriodically(const std::function<void()> &fn,
-                                           boost::posix_time::milliseconds period,
-                                           boost::asio::deadline_timer &timer) {
+void PeriodicalRunner::DoRunFnPeriodically(
+    const std::function<void()> &fn, boost::posix_time::milliseconds period,
+    std::shared_ptr<boost::asio::deadline_timer> timer) {
```

Comment:
It's safer to just pass pointer around, or use the unique pointer of timer to make the ownership clear. I chose the former. Let me know if you think latter is better.